src/main/java/com/kovalevskyi/java/deep/model/graph/Neuron.java

```diff
@@ -27,16 +25,11 @@
      *
      * @param from , Neuron that sends the signal.
      */
-    void forwardSignalReceived(Neuron from, Double value);
-
-    void backwardSignalReceived(Double value);
-
-    void addForwardConnection(Neuron neuron);
-
-    void addBackwardConnection(Neuron neuron, Double weight);
+    public abstract void forwardSignalReceived(Neuron from, Double value);
 
-    Map<Neuron, Double> getBackwardConnections();
+    public abstract void backwardSignalReceived(Double value);
 
-    double getBias();
+    public abstract void connect(Neuron neuron, Double weight);
```

Comment:
The name change feels weird when compared to the unchanged `addBackwardConnection`. If the goal is simplicity, is there a way to simplify the other method too so that they match again?

Reply:
```addBackwardConnection``` is now a package local method and is no longer part of the public API. I would prefer not to have the method at all, but do not know yet how to do it better :(